tests/ui/undocumented_unsafe_blocks.rs

```diff
@@ -490,4 +490,20 @@ unsafe impl CrateRoot for () {}
 // SAFETY: ok
 unsafe impl CrateRoot for (i32) {}
 
+fn issue_9142() {
+    // SAFETY: ok
+    let _ = unsafe {};
+
+    // SAFETY: ok
+    let _ = {
+        if unsafe { true } {
+            todo!();
+        } else {
+            let bar = unsafe {};
```

Comment:
Should this really be seen as ok? And if so, what about something like
```rust
// SAFETY: ok
match foo {
    Foo::Bar => unsafe { something() },
    // A couple hundred lines here
   Foo::Baz => unsafe { something_else() },
}
```

Reply:
How would you define locality? Should I remove blocks?

Reply:
I have removed blocks from the equation.

Reply:
This lint has a problem that it's very much a 'what would the person reading think' sort of thing. It's very formatting dependent.

Even with the blocks ignored a single expression can span a large number of lines.
```rust
some_fn(
    arg1 + with_some_stuff,
    unsafe { arg2.unsafe_fn() },
    some
        .iter()
        .map(|x| do_something(x))
        .collect(),
    //  A few other arguments
    unsafe { arg3.other_unsafe() },
);
```